component/all/all.go

```diff
@@ -2,62 +2,63 @@
+	_ "github.com/grafana/agent/component/otelcol/auth/oauth2"               // Import otelcol.auth.oauth2
+	_ "github.com/grafana/agent/component/otelcol/exporter/jaeger"           // Import otelcol.exporter.jaeger
+	_ "github.com/grafana/agent/component/otelcol/exporter/loki"             // Import otelcol.exporter.loki
+	_ "github.com/grafana/agent/component/otelcol/exporter/otlp"             // Import otelcol.exporter.otlp
+	_ "github.com/grafana/agent/component/otelcol/exporter/otlphttp"         // Import otelcol.exporter.otlphttp
+	_ "github.com/grafana/agent/component/otelcol/exporter/prometheus"       // Import otelcol.exporter.prometheus
+	_ "github.com/grafana/agent/component/otelcol/processor/batch"           // Import otelcol.processor.batch
+	_ "github.com/grafana/agent/component/otelcol/processor/memorylimiter"   // Import otelcol.processor.memory_limiter
+	_ "github.com/grafana/agent/component/otelcol/processor/tail_sampling"   // Import otelcol.processor.tail_sampling
+	_ "github.com/grafana/agent/component/otelcol/receiver/jaeger"           // Import otelcol.receiver.jaeger
+	_ "github.com/grafana/agent/component/otelcol/receiver/kafka"            // Import otelcol.receiver.kafka
+	_ "github.com/grafana/agent/component/otelcol/receiver/loki"             // Import otelcol.receiver.loki
+	_ "github.com/grafana/agent/component/otelcol/receiver/opencensus"       // Import otelcol.receiver.opencensus
+	_ "github.com/grafana/agent/component/otelcol/receiver/otlp"             // Import otelcol.receiver.otlp
+	_ "github.com/grafana/agent/component/otelcol/receiver/prometheus"       // Import otelcol.receiver.prometheus
+	_ "github.com/grafana/agent/component/otelcol/receiver/zipkin"           // Import otelcol.receiver.zipkin
+	_ "github.com/grafana/agent/component/phlare/scrape"                     // Import phlare.scrape
+	_ "github.com/grafana/agent/component/phlare/write"                      // Import phlare.write
+	_ "github.com/grafana/agent/component/prometheus/exporter/apache"        // Import prometheus.exporter.apache
+	_ "github.com/grafana/agent/component/prometheus/exporter/blackbox"      // Import prometheus.exporter.blackbox
+	_ "github.com/grafana/agent/component/prometheus/exporter/consul"        // Import prometheus.exporter.consul
+	_ "github.com/grafana/agent/component/prometheus/exporter/github"        // Import prometheus.exporter.github
+	_ "github.com/grafana/agent/component/prometheus/exporter/mysql"         // Import prometheus.exporter.mysql
+	_ "github.com/grafana/agent/component/prometheus/exporter/process"       // Import prometheus.exporter.process
+	_ "github.com/grafana/agent/component/prometheus/exporter/redis"         // Import prometheus.exporter.redis
+	_ "github.com/grafana/agent/component/prometheus/exporter/unix"          // Import prometheus.exporter.unix
+	_ "github.com/grafana/agent/component/prometheus/kubernetes/podmonitors" // Import prometheus.exporter.apache
```

Comment:
I just want to call out that it's OK for us to merge this PR to make incremental progress, but the component is _not_ currently ready for release, not until the documentation is completed and configGenerator is finished. 

I don't know how we're going to track this (maybe we just remember), but we'll need to remind ourselves to pull the component from the v0.33 release if it's not fully ready by then.